src/app/features/discarded-inscriptions/use-inscribed-spendable-utxos.ts

```diff
@@ -0,0 +1,49 @@
+import { useMemo } from 'react';
+
+import { useNativeSegwitUtxosByAddress } from '@leather.io/query';
+
+import { useCurrentNativeSegwitInscriptions } from '@app/query/bitcoin/ordinals/inscriptions/inscriptions.query';
+import { useCurrentAccountNativeSegwitIndexZeroSigner } from '@app/store/accounts/blockchain/bitcoin/native-segwit-account.hooks';
+import { useDiscardedInscriptions } from '@app/store/settings/settings.selectors';
+
+export function useInscribedSpendableUtxos() {
+  const { hasInscriptionBeenDiscarded } = useDiscardedInscriptions();
+
+  const { data: nativeSegwitInscriptions } = useCurrentNativeSegwitInscriptions();
+
+  const nativeSegwitSigner = useCurrentAccountNativeSegwitIndexZeroSigner();
+  const address = nativeSegwitSigner.address;
+
+  // Utxos but don't filter the inscribed ones
+  const { data: nativeSegwitUtxos } = useNativeSegwitUtxosByAddress({
+    address,
+    filterInscriptionUtxos: false,
+    filterPendingTxsUtxos: true,
+    filterRunesUtxos: true,
+  });
+
+  return useMemo(() => {
+    if (!nativeSegwitUtxos || !nativeSegwitInscriptions) return [];
+
+    // Preformatting utxos so that inscriptions are delcared as an object
+    // property aids the following filter logic
+    const utxosFormatted = nativeSegwitUtxos.map(utxo => ({
+      ...utxo,
+      inscriptions: nativeSegwitInscriptions.filter(
+        inscription => inscription.txid === utxo.txid && Number(inscription.output) === utxo.vout
+      ),
+    }));
+
+    const utxosThatCanBeSpentBecauseAllUtxosInsideWereDiscarded = utxosFormatted
```

Comment:
`utxosThatCanBeSpentBecauseAllUtxosInsideWereDiscarded` what a name!

Reply:
no room for ambiguity !